### PR TITLE
feat: add proguard initial step into Flank configuration

### LINKED ISSUE #1064: R8 And/Or Progaurd
Investigate the use of R8 or Progaurd as a code optimisation, obsfucation and or minimisation tool.

**Additional context**
See more information here: https://jakewharton.com/shrinking-a-kotlin-binary/
Apart of the https://github.com/Flank/flank/issues/774 issue chain.

### PR DESCRIPTION
**Pull request by Sloox**
Fixes #1064 

Adds proguard optimization & minimization onto flank binary. Effectively reducing the binary size to 20MB at time of this pull request. 
This particular pull request does not influence the flank binary size yet due to constraint of: https://github.com/Flank/flank/issues/1072

For now a new task is made and can be run via ./gradlew applyProguard.
It will create and update the flank.jar file, placing it in bash folder under a new name: flank-proguard.jar.  This jar can then be used to 
We done we are waiting for https://github.com/Flank/flank/issues/1072
For now however the basic test cases work for the yaml files.

Again i must reiterate. It does not influence binary size as of yet. However the work was done before the creation of other tasks.

### CODE REVIEW

File: test_runner/proguard/config.pro [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+-keep class ftl.** { *; }
+-keep interface ftl.** { *; }
+
+#google auth
+-keep class com.google.auth.** { *; }
+-keep interface com.google.auth.** { *; }
+
+#ctc wstx
+-keep class com.ctc.wstx.** { *; }
+-keep interface com.ctc.wstx.** { *; }
```
Comment (jan-gogo) [inline test_runner/proguard/config.pro:70]:
```suggestion
-keep interface com.ctc.wstx.** { *; }

```

Comment (jan-gogo) [reply test_runner/proguard/config.pro:70]: There is an option in Intellij that automatically adds an empty line on save but I don't remember the name.

Comment (Sloox) [reply test_runner/proguard/config.pro:70]: Actually have it installed. It maybe didn't apply it as the file type is not supported

File: test_runner/build.gradle.kts [hunk 2]
```diff
@@ -31,6 +31,13 @@ shadowJar.apply {
     manifest {
         attributes(mapOf("Main-Class" to "ftl.Main"))
     }
+    dependencies {
```
Comment (pawelpasterz) [inline test_runner/build.gradle.kts:34]: We have a dependency block at L#186, I assume this is intentional?

Comment (Sloox) [reply test_runner/build.gradle.kts:34]: Yes, it appears that if i put it at L186 it is not working as intended. Hence why it is put here. However with the changes coming with the single gradle project it most likely wont be needed anymore.

Comment (pawelpasterz) [reply test_runner/build.gradle.kts:34]: Interesting, might be due too the fact that Gradle process script from top to bottom?

**Review by pawelpasterz — APPROVED** [review]

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**